### ISSUE TITLE
Create network zone records in non-default project failed

### ISSUE THREAD

**Issue by ec689621** [lxc/lxd#11364]
# Required information

 * Distribution: Ubuntu
 * Distribution version: Focal
 * The output of "lxc info" or if that fails:
   * Kernel version: 5.15.0-58-generic
   * LXC version: 5.10
   * LXD version: 5.10
   * Storage backend in use: zfs, ceph

# Issue description
When setting features.networks.zones=true, try to list/create the records of a network zone in non-default project will complain error " Network zone not found"

While in default project it works. 

# Steps to reproduce

 1. Set features and create zones
```bash
$ lxc project switch pvn
$ lxc project set features.networks.zones=true
$ lxc network zone create pvn.lxd
$ lxc network zone list
+---------+-------------+---------+
|  NAME   | DESCRIPTION | USED BY |
+---------+-------------+---------+
| pvn.lxd |             | 0       |
+---------+-------------+---------+
```
2. Create record in this zone fails
```bash
$ lxc network zone record list pvn.lxd
Error: Network zone not found

$ lxc network zone record create pvn.lxd record1
Error: Network zone not found

$ lxc network zone record list pvn.lxd --project=pvn
Error: Network zone not found
```

3. The same syntax works for the default project
```bash
$ lxc project switch default
$ lxc network zone list
+-------------+-------------+---------+
|    NAME     | DESCRIPTION | USED BY |
+-------------+-------------+---------+
| default.lxd |             | 0       |
+-------------+-------------+---------+

$ lxc network zone record create default.lxd record1
Network zone record record1 created

$ lxc network zone record list default.lxd
+---------+-------------+---------+
|  NAME   | DESCRIPTION | ENTRIES |
+---------+-------------+---------+
| record1 |             |         |
+---------+-------------+---------+
```

# Information to attach

 - [ ] Any relevant kernel output (`dmesg`)
 - [ ] Container log (`lxc info NAME --show-log`)
 - [ ] Container configuration (`lxc config show NAME --expanded`)
 - [ ] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)
 - [ ] Output of the client with --debug
 - [ ] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)

**Comment by tomponline** [comment]
You're absolutely right, the zone record handlers are not using the correct project. I will fix this ready for LXD 5.11 that is being released later this week.

**Comment by tomponline** [comment]
Also spotted another bug in the zones record DB schema as part of this so have fixed that too.
